File: plugin/src/main/java/copilotj/awt/component/LabelNode.java
/**
 * SPDX-FileCopyrightText: Copyright contributors to the CopilotJ project.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

package copilotj.awt.component;

import java.awt.Component;
import java.awt.Label;

public class LabelNode extends AbstractComponentNode<Label> {
  public static class Provider implements ComponentNodeProvider {
    @Override
    public ComponentNode tryCreate(final Component component) {
      if (component instanceof Label) {
        return new LabelNode((Label) component);
      }
      return null;
    }
  }

  public final String text;

  public LabelNode(final Label component) {
    super("java.awt.Label", component);
    this.text = component.getText() != null ? component.getText().trim() : null;
  }

  @Override
  public String describe() {
    return "Label: text=" + text;
  }
}

File: copilotj/plugin/awt/component/label_node.py
# SPDX-FileCopyrightText: Copyright contributors to the CopilotJ project.
#
# SPDX-License-Identifier: Apache-2.0

from typing import Literal, override

from copilotj.plugin.awt._base import ComponentBase

__all__ = ["LabelNode"]


class LabelNode(ComponentBase[Literal["java.awt.Label"]]):
    text: str

    @override
    def role(self) -> str:
        return "label"

    @override
    def _state_inline(self) -> str | None:
        # Raw, unquoted text (sanitized to one line, truncated); renders as
        # `- label: {text}`. Labels carry no ref and no actions.
        text = self.text if self.text is not None else "<empty>"
        text = text.replace("\n", " ").replace("\r", " ").replace("\t", " ")
        if len(text) > 300:
            text = text[:300] + "..."
        return text
